packages/neuron-wallet/src/services/transaction-sender.ts

```diff
@@ -183,13 +189,32 @@ export default class TransactionSender {
 
     const lockHashes = new Set(witnessSigningEntries.map(w => w.lockHash))
 
+    const metadata = {
+      locks: {
+        skipped: new Set<string>(),
+      },
+      skipLastInput,
+    }
+
     for (const lockHash of lockHashes) {
       const witnessesArgs = witnessSigningEntries.filter(w => w.lockHash === lockHash)
-      // A 65-byte empty signature used as placeholder
-      witnessesArgs[0].witnessArgs.setEmptyLock()
 
       const privateKey = findPrivateKey(witnessesArgs[0].lockArgs)
 
+      if (!privateKey) {
+        metadata.locks.skipped.add(lockHash)
+        witnessSigningEntries.forEach((entry, idx) => {
+          if (entry.lockHash === lockHash) {
+            const rawWitness = tx.witnesses[idx]
```

Comment:
Could the `tx.witnesses[idx]` be a `''` instead `'0x'`? If we are not sure about it, maybe we can replace this line with

```ts
const rawWitness = tx.witnesses[idx] || '0x'
```

Reply:
This branch is for skipped input/witness pair, and if the witness is a string, it will be returned without modification. From my perspective, it's weird to replace `''` with `'0x'` because it does some work but it's meaningless.

Reply:
How would you think of the explanation

Reply:
> This branch is for skipped input/witness pair, and if the witness is a string, it will be returned without modification. From my perspective, it's weird to replace '' with '0x' because it does some work but it's meaningless.

The reason I mentioned this is because I tried to skip unidentified locks locally but failed due to an empty [witness `""`](https://github.com/nervosnetwork/neuron/commit/27fcc6bd2dc45148b5f537af6d62abb231574e02#diff-6b4e52aade78db37f085688194f39382d8411f142f051ac17a199f52f6953babR248) in witnesses. And I was not quite sure which was wrong. If you haven't encountered the issue, maybe we don't need a default witness placeholder